serve/filters/filter/tx_test.go

```diff
@@ -0,0 +1,223 @@
+package filter
+
+import (
+	"sync"
+	"testing"
+
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetHashAndChanges(t *testing.T) {
+	t.Parallel()
+
+	txs := []*types.TxResult{
+		{Tx: []byte(`c25dda249cdece9d908cc33adcd16aa05e20290f`)},
+		{Tx: []byte(`71ac9eed6a76a285ae035fe84a251d56ae9485a4`)},
+		{Tx: []byte(`356a192b7913b04c54574d18c28d46e6395428ab`)},
+		{Tx: []byte(`da4b9237bacccdf19c0760cab7aec4a8359010b0`)},
+		{Tx: []byte(`77de68daecd823babbb58edb1c8e14d7106e83bb`)},
+		{Tx: []byte(`1b6453892473a467d07372d45eb05abc2031647a`)},
+		{Tx: []byte(`ac3478d69a3c81fa62e60f5c3696165a4e5e6ac4`)},
+		{Tx: []byte(`c1dfd96eea8cc2b62785275bca38ac261256e278`)},
+	}
+
+	// create a new tx filter
+	f := NewTxFilter()
+	assert.Equal(t, TxFilterType, f.GetType())
+
+	for _, tx := range txs {
+		f.UpdateWithTx(tx)
+	}
+
+	// get the hashes of the txs
+	hashes := f.GetHashes()
+	for i, tx := range txs {
+		assert.Equal(t, tx.Tx.Hash(), hashes[i])
+	}
+
+	// get the chages from the filter
+	changes := f.GetChanges().([]*types.TxResult)
+	require.Len(t, changes, len(txs))
+
+	for i, tx := range txs {
+		assert.Equal(t, tx, changes[i])
+	}
+
+	assert.Empty(t, f.txrs)
+}
+
+var txs = []*types.TxResult{
```

Comment:
Please move this into the test itself, instead of leaving it as a package-global 🙏